src/mumble/API_v_1_x_x.cpp

```diff
@@ -1093,6 +1172,80 @@ void MumbleAPI::requestSetLocalUserComment_v_1_0_x(mumble_plugin_id_t callerID,
 	EXIT_WITH(MUMBLE_STATUS_OK);
 }
 
+void MumbleAPI::requestLinkChannels_v_1_3_x(mumble_plugin_id_t callerID, mumble_channelid_t *channelList,
+											std::size_t channelCount, std::shared_ptr< api_promise_t > promise) {
+	if (QThread::currentThread() != thread()) {
+		// Invoke in main thread
+		QMetaObject::invokeMethod(this, "requestLinkChannels_v_1_3_x", Qt::QueuedConnection,
+								  Q_ARG(mumble_plugin_id_t, callerID), Q_ARG(mumble_channelid_t *, channelList),
+								  Q_ARG(std::size_t, channelCount), Q_ARG(std::shared_ptr< api_promise_t >, promise));
+
+		return;
+	}
+	api_promise_t::lock_guard_t guard = promise->lock();
+	if (promise->isCancelled()) {
+		return;
+	}
+
+	VERIFY_PLUGIN_ID(callerID);
+
+	if (channelCount < 2) {
+		EXIT_WITH(MUMBLE_STATUS_OK);
+	}
+
+	const Channel *channel = Channel::get(static_cast< unsigned int >(channelList[0]));
+
+	if (!channel) {
+		EXIT_WITH(MUMBLE_EC_CHANNEL_NOT_FOUND);
+	}
+
+	for (size_t i = 1; i < channelCount; i++) {
+		const Channel *link = Channel::get(static_cast< unsigned int >(channelList[i]));
+		if (!link) {
+			EXIT_WITH(MUMBLE_EC_LINK_CHANNEL_NOT_FOUND);
+		}
+		Global::get().sh->addChannelLink(channel->iId, link->iId);
+	}
```

Comment:
I think we should split this into two loops. In the first round we validate all provided channels and only in the second iteration will we actually perform the linking. This way we avoid completing the job halfway before realizing that one of the channel IDs is invalid and aborting.

Also, I think we should check whether the current channel is already in the set of linked channels for `channel` and only link channels if they are not already linked.

Finally, there should be a new function inside `ServerHandler` that can transmit all to-be-linked channels in a single message. Otherwise, we risk running into the server's (per default rather strict) rate limiter and request messages will get dropped.
You can essentially copy the implementation of `addChannelLink` and only add change the parameter to a `std::vector<unsigned int>` and then `add_links_add` in a loop. (Also you then can make the existing `addChannelLink` call the more general function in order to have a shared implementation between the two.